internal/locate/region_request.go

```diff
@@ -604,7 +605,7 @@ func (state *accessFollower) isCandidate(idx AccessIndex, replica *replica) bool
 		// The request can only be sent to the leader.
 		((state.option.leaderOnly && idx == state.leaderIdx) ||
 			// Choose a replica with matched labels.
-			(!state.option.leaderOnly && (state.tryLeader || idx != state.leaderIdx) && replica.store.IsLabelsMatch(state.option.labels))) &&
+			(!state.option.leaderOnly && (state.tryLeader || idx != state.leaderIdx) && (state.lastIdx >= 0 || replica.store.IsLabelsMatch(state.option.labels)))) &&
 		// Make sure the replica is not unreachable.
 		replica.store.getLivenessState() != unreachable
```

Comment:
It's better to split this check into different checks like the master branch here like:
```
func (state *accessFollower) isCandidate(idx AccessIndex, replica *replica) bool {
   if condition1 not match 
        return false
   if condition2 not match 
        return false
   if condition3 matches
        return true
   return false    
}
``` 
, it's quite difficult to maintain.

Reply:
Done, and I did this for master branch before.